AlwaysOnKotlin/views/src/main/java/com/example/android/wearable/wear/alwayson/MainActivity.kt

```diff
@@ -188,16 +188,20 @@ class MainActivity : FragmentActivity(), AmbientModeSupport.AmbientCallbackProvi
      * (active mode = coroutines and ambient mode = Alarm).
      */
     private fun refreshDisplayAndSetNextUpdate() {
+        Log.d(TAG, "refreshDisplayAndSetNextUpdate()")
         loadDataAndUpdateScreen()
         val instant = Instant.now(clock)
-        if (ambientController.isAmbient) {
+        if (ambientCallbackState.isAmbient) {
             val triggerTime = instant.getNextInstantWithInterval(AMBIENT_INTERVAL)
-            ambientUpdateAlarmManager.setExact(
-                AlarmManager.RTC_WAKEUP,
-                triggerTime.toEpochMilli(),
-                ambientUpdatePendingIntent
-            )
+            if (Build.VERSION.SDK_INT < 33) {
```

Comment:
This piece is a bit confusing, what we suggest to do for app targeting >=33 (Wear 4)?
1) Keep using exact alarm (https://developer.android.com/training/scheduling/alarms#exact)
2) Not override default interval and adhere to standard frequency (1 min)
3) Use inexact alarm e.g. `setWindow(..)` (more info at https://developer.android.com/training/scheduling/alarms#use-cases)

(1) doesn't seem like a great idea because of permission and outlined use cases suggests that our case might not require exact alarm (https://developer.android.com/training/scheduling/alarms#use-cases).

We may want to try (3) with using `setWindow` or `setAndAllowWhileIdle()` if this will work as intended for this use case, otherwise fall back to (2)?

Reply:
Do you mind if I land and follow up with a change focused specifically on this part?

I agree with your suggestions, just want a separate PR.

Reply:
Yes, this worth a separate PR/discussion wrt targetSDK bump.